nixos/modules/virtualisation/qemu-vm.nix

```diff
@@ -942,7 +957,8 @@ in
             # Sync with systemd's tmp.mount;
             options = [ "mode=1777" "strictatime" "nosuid" "nodev" "size=${toString config.boot.tmpOnTmpfsSize}" ];
           };
-
+      } else {}) //
+      {
         "/nix/${if cfg.writableStore then ".ro-store" else "store"}" =
           mkIf cfg.useNixStoreImage
```

Comment:
Might change this to `optionalAttrs` for consistency.

Reply:
You mean also setting the `mkIf` entries with `optionalAttrs [...] { } //`? I guess that could change the value in some cases since `mkIf` still clears the user-supplied value if the condition is false but I don't know if that would break anything. And it would make things more customizable which is good.